sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -668,33 +668,32 @@ public static string DetermineRegion(string url)
         /// </returns>
         public static string DetermineService(string url)
         {
-            int delimIndex = url.IndexOf("//", StringComparison.Ordinal);
-            if (delimIndex >= 0)
-                url = url.Substring(delimIndex + 2);
+            var urlSpan = url.AsSpan();
 
-            string[] urlParts = url.Split(new char[] { '.' }, StringSplitOptions.RemoveEmptyEntries);
-            if (urlParts == null || urlParts.Length == 0)
-                return string.Empty;
+            var doubleSlashIndex = urlSpan.IndexOf(DoubleSlash, StringComparison.Ordinal);
+            if (doubleSlashIndex >= 0)
+                urlSpan = urlSpan.Slice(doubleSlashIndex + 2);
 
-            string servicePart = urlParts[0];
-            int hyphenated = servicePart.IndexOf('-');
-            string service;
-            if (hyphenated < 0)
-            { service = servicePart; }
-            else
-            { service = servicePart.Substring(0, hyphenated); }
+            var dotIndex = urlSpan.IndexOf('.');
 
-            // Check for SQS : return "sqs" incase service is determined to be "queue" as per the URL.
-            if (service.Equals("queue"))
-            {
-                return "sqs";
-            }
-            else
+            if (dotIndex < 0)
+                return string.Empty;
+
+            var servicePartSpan = urlSpan.Slice(0, dotIndex);
+            var hyphenIndex = servicePartSpan.IndexOf('-');
+            if (hyphenIndex > 0)
             {
-                return service;
+                servicePartSpan = servicePartSpan.Slice(0, hyphenIndex);
             }
+
+            // Check for SQS : return "sqs" in case service is determined to be "queue" as per the URL.
+            return servicePartSpan.Equals(Queue, StringComparison.OrdinalIgnoreCase) ? "sqs" : servicePartSpan.ToString();
         }
 
+        // Compiler trick to directly refer to static data in the assembly
+        private static ReadOnlySpan<char> DoubleSlash => new[] { '/', '/' };
```

Comment:
@stevenaw you might enjoy this ;)

Reply:
That's a neat trick, thanks! I'll have to look out for opportunities to use this